### PR TITLE
Add JSON logging

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
@jcollins-g – I was pondering that this case might be nice to export everything to stdout – just so it's easy to pipe to a file or another process.

...let's discuss

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 0.14.2
```
Comment (jcollins-g) [inline CHANGELOG.md]: remove version number from CL; you don't know what version this will be published as.

File: bin/dartdoc.dart [hunk 2]
```diff
@@ -122,48 +123,69 @@ main(List<String> arguments) async {
...
       } else {
-        print(message);
+        output['message'] = record.message;
       }
-    } else {
-      stderr.writeln(message);
-    }
-  });
+
+      print(JSON.encode(output));
```
Comment (jcollins-g) [inline bin/dartdoc.dart]: Writing this to stdout is OK if you can redirect stdout internally in dartdoc to something else when writing json, so an errant print statement somewhere doesn't break your JSON.

Indeed, that's how this should be implemented generally, assuming that's possible in dart.  That way you don't have to require that all output goes through this library.

File: CHANGELOG.md (hunk 1 above)
Comment (kevmoo) [reply CHANGELOG.md]: So this is the pattern we've used through packages.

First change post-release: bump version & -dev

So 1.2.3 becomes 1.2.4-dev

The changelog entries follow:

```
## 1.2.4
```

Then if it becomes a breaking release...

1.2.4-dev become 2.0.0-dev

and update the changelog accordingly

That's how we (try) to manage all packages.

**Comment by kevmoo** [comment]
> Request adding an end-to-end test of this feature that validates good JSON is written when generating docs for the test package.

On it!

**Comment by kevmoo** [comment]
@jcollins-g – PTAL

File: CHANGELOG.md (hunk 1 above)
Comment (jcollins-g) [reply CHANGELOG.md]: ok, didn't know that was a standard practice as well.  we'll try and do it that way then.

**Comment by jcollins-g** [comment]
This looks pretty good now.  Use zones to redirect errant print statements as discussed and then this will be fine.

File: CHANGELOG.md (hunk 1 above)
Comment (devoncarew) [reply CHANGELOG.md]: Not to muddy the waters, but I use `## unreleased` as a placeholder.

Comment (kevmoo) [reply CHANGELOG.md]: done

**Comment by kevmoo** [comment]
PTAL

**Pull request merged**